Sources/ParseLiveQuery/Client.swift

```diff
@@ -149,13 +150,13 @@ extension Client {
             )
             subscriptions.append(subscriptionRecord)
 
-            if socket == nil {
-                if !disconnected {
-                    reconnect()
-                }
-            } else {
-                sendOperationAsync(.Subscribe(requestId: subscriptionRecord.requestId, query: query))
-            }
+        if socket == nil || socket?.readyState == .CLOSED{
+            userDisconnected = false
```

Comment:
All of this is a great idea!
Feel like we can do the last 2 even in this pull request or pretty easily?